From 83ba50c1873c637ce040f1950c05224639dc5811 Mon Sep 17 00:00:00 2001
From: Luc Bonnafoux <luc.bonnafoux@ssi.gouv.fr>
Date: Thu, 12 Jun 2025 15:18:34 +0200
Subject: [PATCH 1/5] usb/auth: Add compilation options

- Add `USB_AUTH` compilation option in the usb Kconfig.
- Activate the option in the x86_64-softmmu target.

Co-authored-by: Luc Bonnafoux <luc.bonnafoux@ssi.gouv.fr>
Co-authored-by: Nicolas Bouchinet <nicolas.bouchinet@ssi.gouv.fr>
---
 configs/targets/x86_64-softmmu.mak | 2 ++
 hw/usb/Kconfig                     | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/configs/targets/x86_64-softmmu.mak b/configs/targets/x86_64-softmmu.mak
index c5f882e5ba..fa851a0e25 100644
--- a/configs/targets/x86_64-softmmu.mak
+++ b/configs/targets/x86_64-softmmu.mak
@@ -4,3 +4,5 @@ TARGET_SUPPORTS_MTTCG=y
 TARGET_NEED_FDT=y
 TARGET_KVM_HAVE_GUEST_DEBUG=y
 TARGET_XML_FILES= gdb-xml/i386-64bit.xml
+
+CONFIG_USB_AUTH=y
\ No newline at end of file
diff --git a/hw/usb/Kconfig b/hw/usb/Kconfig
index f569ed7eea..276df8a5f9 100644
--- a/hw/usb/Kconfig
+++ b/hw/usb/Kconfig
@@ -1,6 +1,11 @@
 config USB
     bool
 
+config USB_AUTH
+    bool
+    default n
+    select USB
+
 config USB_UHCI
     bool
     default y if PCI_DEVICES
-- 
2.50.0

